packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -278,7 +278,7 @@ class IOSDevice extends Device {
 
       Future<int> launch = runCommandAndStreamOutput(launchCommand, trace: true);
 
-      List<Uri> uris = await launch.then((int result) async {
+      List<Uri> uris = await launch.then<List<Uri>>((int result) async {
```

Comment:
Is there any chance that this is a holdover from dev.10.0?  We had a regression related to async/await, but I can't reproduce this error in either dev.10.1 or bleeding edge.

I'm not testing the actual code, so it's possible there's an issue I'm not seeing, but I have a repro that should be capturing everything relevant and I don't see any issues.